test/test_serialization.py

```diff
@@ -279,6 +279,16 @@ def test_serialization_offset_gzip(self):
         self.assertTrue(torch.equal(a, b))
         self.assertEqual(i, j)
 
+    @unittest.skipIf(IS_WINDOWS, "NamedTemporaryFile on windows")
+    def test_serialization_sparse(self):
+        x = torch.zeros(3, 3)
+        x[1][1] = 1
+        x = x.to_sparse()
+        with tempfile.NamedTemporaryFile() as f:
+            torch.save(x, f.name)
+            y = torch.load(f.name)
+            self.assertEqual(x, y)
+
```

Comment:
Can we add a failing test-case here with wrongly constructed values, calling the internal constructor?

Reply:
That was a very good call! I found a bug that way!